index.html

```diff
@@ -207,256 +242,549 @@ <h2>Les tableaux</h2>
 					<pre>
-						<code>
-// avec ===, fini de jouer : vérif valeur ET TYPE !
-
-42 === '42'        // => false
-null === undefined // => false
-null === 0         // => false
-0 === undefined    // => false
-0 === false        // => false
-'0' === false      // => false
-NaN === NaN        // => false -- rien à faire !
+						<code class="javascript">
+/// Fonction standard
+function a() {
+	/// this = contexte appelant
+}
+/// Variable fonction anonyme (ne pas utiliser)
+const b = function() {
+	/// this = contexte appelant
+}
+/// Variable fonction nommée (usage assez rare)
+const c = function c() {
+	/// this = contexte appelant
+}
+/// Lambda / "fat arrow" (anonyme)
+const d = () => {
+	/// this = contexte de la déclaration
+}
+/// fonction auto-appelante (usage particulier)
+(function(){
+	/// this = contexte appelant
+})()
 						</code>
 					</pre>
 				</section>
 				<section>
-					<h2>in et delete</h2>
-					<pre>
-						<code class='javascript'>
+						<h1>Déclaration</h1>
+						<pre>
+							<code class="javascript">
+/// La syntaxe de a est un raccourci pour celle de b. (usage assez rare)
+const e {
+    a() {
+        /// this = contexte appelant
+    },
+    b: function b() {
+        /// this = contexte appelant
+    }
+}
+/// Dans une classe
+class G {
+    a() {
+        /// this = contexte appelant
+		}
+		/// Variable d'instance
+    b = () => {
+        /// this = instance de la classe
+    }
+}
+							</code>
+						</pre>
+					</section>
+				<section>
+					<h1>Variables</h1>
+					<p>Il existe trois façons de déclarer des variable en javascript:</p>
+					<ul>
+						<li>var: On déclare une variable, éventuellement en initialisant sa valeur.</li>
+						<li>let: On déclare une variable dont la portée est celle du bloc courant, éventuellement en initialisant sa valeur.</li>
+						<li>const: On déclare une constante nommée, dont la portée est celle du bloc courant, accessible en lecture seule.</li>
+					</ul>
+					<blockquote>var n'est plus vraiment utilisé dans les nouvelles normes. On préfèrera utiliser const dans les 3/4 des cas, et let dans les cas ou la variables est réaffectable.</blockquote>
+				</section>
+				<section>
+						<h1>Boucles</h1>
+						<p>Depuis ECMAScript2015 le prototype ARRAY a gagné de nouvelles méthode d'itération sur les éléments. Nous allons en présenter deux ici:</p>
+						<table>
+							<tr>
+								<td>map</td>
+								<td>immutable</td>
+								<td>Crée un nouveau tableau avec les résultats de l'appel d'une fonction fournie sur chaque élément du tableau appelant.</td>
+							</tr>
+							<tr>
+								<td>foreach</td>
+								<td>mutable</td>
+								<td>Permet d'exécuter une fonction donnée sur chaque élément du tableau. Ne renvoie rien</td>
+							</tr>
+						</table>
+				</section>
+				<section>
+						<h1>Boucles</h1>
+						<pre>
+								<code class='javascript'>
+const ARRAY = [1,2,3,4];
+for(let i = 0, _l=ARRAY.length; i < _l; i++ ){console.log(ARRAY[i])}
```

Comment:
Si tu veux rester cohérent avec les deux suivants

```suggestion

for(let i = 0, _l=ARRAY.length; i < _l; i++ ){console.log(ARRAY[i], i)}
```